client/gutenberg/extensions/map/save.js

```diff
@@ -19,16 +19,16 @@ class MapSave extends Component {
 				</li>
 			);
 		} );
-		// All camelCase attribute names converted to snake_case data attributes
+		// All camelCase attribute names converted to kebab-case data attributes
 		return (
 			<div
 				className={ className }
-				data-map_style={ mapStyle }
-				data-map_details={ mapDetails }
+				data-map-style={ mapStyle }
+				data-map-details={ mapDetails }
 				data-points={ JSON.stringify( points ) }
 				data-zoom={ zoom }
-				data-map_center={ JSON.stringify( mapCenter ) }
-				data-marker_color={ markerColor }
+				data-map-center={ JSON.stringify( mapCenter ) }
+				data-marker-color={ markerColor }
```

Comment:
It really seems like we should be able to automate this better. This puts data into a structure that will be used by `FrontendManagement`, is that correct? It seems like we should have a utility, perhaps just some methods, that put data into and read data from data-attributes in a consistent way.

This bit here could become something like the following rather than manually writing the keys:

```jsx
import { compose, mapKeys, kebabCase } from 'lodash';

function objectToDataAttributes( obj ) {
  return mapKeys( obj, ( _ /* value */, key ) => `data-${ kebab-case( key ) }` );
}

function stringifyValues( obj ) {
  return mapValues( obj, JSON.stringify );
}

const toSpreadableDataAtts = compose( objectToDataAttributes, stringifyValues );

<div
  className={ className, kebabCase }
  { ...toSpreadableDataAtts( attributes ) }
/>
```

Reading would need to `JSON.parse` all the attributes values in `FrontendManagement`, a symmetry I like and would save some conditional handling for different types.

Just some thoughts. Not everything is stringify-able, which would be a limitation. This could be good work for a next version.

Reply:
I had originally wanted a solution like this too, and still would like to add at some point. Here are a few reasons why I didn't pursue, at least for now:

1) This idea of passing block attributes through to the view script by way of data attributes is new and somewhat difficult to describe concisely, so I liked the idea of a highly readable `save()` function at least for the first application of the pattern.
2) I didn't want to incur the cost of JSON.stringify()-ing non-object values, although this is so trivial this was probably a ridiculous concern.
3) I suspected there might be some attribute values that required additional pre-processing before serialization, and wanted to keep that option easily open. In the end I didn't need this.
4) I also anticipate that there might be attributes which we needed _only_ in editor, and didn't want them serialized/de-serialized unnecessarily. This also proved false for the map block.

Anyway, I'd propose deferring this for now for fear it will introduce new issues, and taking a pass at it after v1 of this block ships. Sound OK?

Reply:
> I'd propose deferring this for now for fear it will introduce new issues

Yes, please don't include any of these suggestions in this release 🙂

Reply:
OK great. I'll prepare a PR with this shortly.